csrc/scheduler/reduction_utils.cpp

```diff
@@ -1009,5 +1003,30 @@ std::ostream& operator<<(std::ostream& os, ReductionType reduction_type) {
   return os;
 }
 
+void sharedMemoryConsumerVectorization(
+    std::vector<TensorView*>& smem_consumers) {
+  // Vectorization of smem consumers, they were created with cacheAfter().
+  // Can't directly use the vectorization factor set for inputs due to
+  // potential different data types, e.g. fp16 inputs and fp32 smem_consumers.
+  // When this happens, there are two additional optimizations should be done:
+  // (1) writing to smem should be vectorized.
+  // (2) when n_loads > 1, still has bank conflicts.
+  // See test SharedMemoryPersistentVectFactor.
+  for (auto tv : smem_consumers) {
+    NVF_ERROR(
+        tv->definition()->isA<LoadStoreOp>(),
+        "smem consumers should be LoadStoreOp. Got: ",
+        tv->definition()->toString());
+    auto innermost_extent = tv->axis(-1)->extent()->evaluate().as<int64_t>();
```

Comment:
Will add a check to ensure `innermost_extent == vectorization factor set for cached input`

Reply:
Can you please elaborate? Why not added yet?

Reply:
added. I was waiting for the CI to complete, didn't want to re-start it due to a small change,